validator/tests/test_scheduler/test_schedulers_with_yaml.py

```diff
@@ -285,6 +285,23 @@ def test_complex_dependency_failures(self):
             context_manager=context_manager,
             name='complex_dependency_failures.yaml')
 
+    def test_single_txn_per_batch(self):
```

Comment:
How does this test prevent a regression for this issue?  It's not at all clear from the test case itself. 

Can you speak a bit more about these items?

Reply:
Yeah, sorry the explanation was in the file I forgot to commit...